crates/fs/private/namefilter/namefilter.rs

```diff
@@ -0,0 +1,75 @@
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+use super::BloomFilter;
+
+//--------------------------------------------------------------------------------------------------
+// Constants
+//--------------------------------------------------------------------------------------------------
+
+pub const SATURATION_THRESHOLD: usize = 1019;
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+/// Namefilters are 2048-bit bloom filters.
+///
+/// In WNFS they represent the identity of a file or directory, doubling as a store for checking the ancestor of the file or directory.
+pub type Namefilter = BloomFilter<256, 30>;
+
+//--------------------------------------------------------------------------------------------------
+// Implementations
+//--------------------------------------------------------------------------------------------------
+
+impl Namefilter {
+    /// Creates a saturated namefilter.
+    pub fn new_saturated() -> Self {
+        let mut filter = Self::new();
+        filter.saturate();
+        filter
+    }
```

Comment:
Hm, I see how this is useful for tests, but it's not useless for any application. An empty, but saturated namefilter is essentially useless: You can't do anything with it anymore and it doesn't represent any information.

Reply:
Yeah. Shouldn't have been there.